shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/resetpassword/ResetPasswordViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package co.nimblehq.blisskmmic.presentation.modules.resetpassword
+
+import co.nimblehq.blisskmmic.MR
+import co.nimblehq.blisskmmic.data.network.helpers.toErrorMessage
+import co.nimblehq.blisskmmic.domain.usecase.ResetPasswordUseCase
+import co.nimblehq.blisskmmic.presentation.model.NotificationUiModel
+import co.nimblehq.blisskmmic.presentation.modules.BaseViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+data class ResetPasswordViewState(
+    val successNotification: NotificationUiModel? = null,
+    val isLoading: Boolean = false,
+    val error: String? = null
+) {
+    constructor() : this(null, false, null)
+    constructor(error: String?) : this(null, false, error)
+}
+
+class ResetPasswordViewModel(
+    private val resetPasswordUseCase: ResetPasswordUseCase,
```

Comment:
```suggestion
    private val resetPasswordUseCase: ResetPasswordUseCase
```

Reply:
I included this because Android Studio will create weird indent without it. Anything familiar you can think of? 🤔 

https://user-images.githubusercontent.com/6356137/205035752-d1d54ea6-ca09-4116-a6ee-8c1fc4080ed9.mov

Reply:
@blyscuit hmm I see, then since there is only one attribute for this class, I think we can also refactor it to 😁
```
class ResetPasswordViewModel(private val resetPasswordUseCase: ResetPasswordUseCase) :
    BaseViewModel() {
```